src/server/services/chatService/cache.js

```diff
@@ -15,8 +15,11 @@ export async function getUserId(userHandle) {
 }
 
 export async function getChannelId(channelName) {
-  const channelIdMap = await _getChannelIdMap()
-  return channelIdMap.get(channelName)
+  return memoryCache.wrap(channelName, async () => {
+    const channelMap = await _getChannelIdMapUncached()
+    await Promise.each(channelMap.keys(), name => memoryCache.set(name, channelMap.get(name)))
```

Comment:
Ok awesome, yeah I think I put myself in a corner in trying to preserve as much of the pre-existing code. I would definitely prefer to rewrite the bulk of this file.